### PR TITLE
Cleaning up add-import finding code.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Overall changes here:

1. Made naming more consistent in this file.
2. Added comments to make purpose of some methods clearer.
3. Inlined a lot of methods that were only called in one place.
4. Ordered methods to keep extracted methods closer to their callsite.
5. Moved nuget related searching to a new file.

**Comment by CyrusNajmabadi** [comment]
Prremptively merging in.  Let me know if you have any issues.

File: src/Features/Core/Portable/AddImport/SymbolReferenceFinder.cs [hunk 1]
```diff
@@ -195,6 +195,38 @@ private ImmutableArray<SymbolReference> DeDupeAndSortReferences(ImmutableArray<S
...
+                    return ImmutableArray<SymbolReference>.Empty;
+                }
+
+                if (ExpressionBinds(nameNode, checkForExtensionMethods: false, cancellationToken: searchScope.CancellationToken))
+                {
+                    return ImmutableArray<SymbolReference>.Empty;
+                }
+
+                var symbols = await searchScope.FindDeclarationsAsync(name, nameNode, SymbolFilter.Namespace).ConfigureAwait(false);
+                var namespaceSymbols = OfType<INamespaceSymbol>(symbols);
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/SymbolReferenceFinder.cs]: Unused?

File: src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs [hunk 2]
```diff
@@ -90,7 +91,7 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
                         var allSymbolReferences = await FindResultsAsync(document, semanticModel, diagnostic, node, cancellationToken).ConfigureAwait(false);
 
                         // Nothing found at all. No need to proceed.
-                        if (allSymbolReferences == null || allSymbolReferences.Count == 0)
+                        if (allSymbolReferences.Length == 0)
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: Use `IsDefaultOrEmpty` to protect against uninitialized values?

File: src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs [hunk 3]
```diff
@@ -123,7 +124,7 @@ internal abstract partial class AbstractAddImportCodeFixProvider<TSimpleNameSynt
 
             // Look for exact matches first:
             var exactReferences = await FindResultsAsync(projectToAssembly, referenceToCompilation, project, finder, exact: true, cancellationToken: cancellationToken).ConfigureAwait(false);
-            if (exactReferences?.Count > 0)
+            if (exactReferences.Length > 0)
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: Any chance of a default value here?

File: src/Features/Core/Portable/AddImport/SymbolReferenceFinder.cs [hunk 4]
```diff
@@ -312,10 +312,22 @@ private bool HasAccessibleStaticFieldOrProperty(INamedTypeSymbol namedType, stri
                 ImmutableArray<SymbolResult<IMethodSymbol>> methodSymbols,
                 SyntaxNode expression, CancellationToken cancellationToken)
             {
+                return GetViableExtensionMethodsWorker(methodSymbols, cancellationToken).WhereAsArray(
+                    s => _owner.IsViableExtensionMethod(s.Symbol, expression, _semanticModel, _syntaxFacts, cancellationToken));
+            }
+            private ImmutableArray<SymbolResult<IMethodSymbol>> GetViableExtensionMethods(
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/SymbolReferenceFinder.cs]: Add blank line

File: src/Features/Core/Portable/AddImport/SymbolReferenceFinder.cs [hunk 5]
```diff
@@ -312,10 +312,22 @@ private bool HasAccessibleStaticFieldOrProperty(INamedTypeSymbol namedType, stri
...
+                ImmutableArray<SymbolResult<IMethodSymbol>> methodSymbols,
+                ITypeSymbol typeSymbol, CancellationToken cancellationToken)
+            {
+                return GetViableExtensionMethodsWorker(methodSymbols, cancellationToken).WhereAsArray(
+                    s => _owner.IsViableExtensionMethod(s.Symbol, typeSymbol));
+            }
+            private ImmutableArray<SymbolResult<IMethodSymbol>> GetViableExtensionMethodsWorker(
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/SymbolReferenceFinder.cs]: Add blank line.

**Pull request merged**